src/reporting.rs

```diff
@@ -0,0 +1,78 @@
+use anyhow::Result;
+use reqwest::Client;
+use tracing::debug;
+
+use crate::models::{
+    DiscordEmbed, DiscordEmbedAuthor, DiscordWebhookEvent, MemberCreator, WebhookActionDisplay,
+    WebhookEvent,
+};
+
+pub struct DiscordReporter {
+    endpoint: String,
+    client: Client,
+}
+
+impl DiscordReporter {
+    pub fn new(endpoint: String) -> Self {
+        Self {
+            endpoint,
+            client: Client::new(),
+        }
+    }
+
+    pub async fn report(&self, event: WebhookEvent) -> Result<()> {
+        if Self::should_skip(&event) {
+            debug!("Skipping event");
```

Comment:
Maybe it's useful to log the event so that we can find out what we have skipped?
```suggestion
            debug!("Skipping event {:?}", event);
```